--- verapdf-cliOLD	2021-01-27 00:45:30.751075515 -0400
+++ verapdf-cli	2021-01-27 00:51:02.788057898 -0400
@@ -118,22 +118,23 @@
   [ -n "$REPO" ] && REPO=`cygpath --path --windows "$REPO"`
 fi
 
-# This sets a scaling factor for the veraPDF GUI window layout.
-# You should only change this if there is a problem with the window layout on your machine.
-# The usual cause of such problems is a high resolution display, greater than 1920x1080 and use if Windows text scaling for such displays.
-# The root cause us that Java Swing applications are rendered based on pixels, see this issue for more details: http://openjdk.java.net/jeps/263
-# If you have problems with a cramped window and invisible window controls you can increase the SCALE_FACTOR below to 1.5 or even 2.0
-SCALE_FACTOR=1.0
+# Allow Java CreateTempFile API to work under flatpak
+# Needs flatpak "--filesystem=xdg-cache" or "--filesystem=home"
+export JAVA_TOOL_OPTIONS="-Djava.io.tmpdir=${XDG_CACHE_HOME}/tmp"
+
+# We saved original config files in read-only location 'config-default'
+# copy them to 'config' writable dir before launching app
+# This workarounds known flatpak limitation, see https://github.com/flatpak/flatpak/issues/545#issuecomment-278931352
+cp -u /app/verapdf/config-default/* /var/config/
 
 exec "$JAVACMD" $JAVA_OPTS  \
   -classpath "$CLASSPATH" \
   -Dfile.encoding="UTF8" \
   -XX:+IgnoreUnrecognizedVMOptions \
-  -Dapp.name="VeraPDF validation GUI" \
+  -Dapp.name="VeraPDF validation CLI" \
   -Dapp.pid="$$" \
   -Dapp.repo="$REPO" \
   -Dapp.home="$BASEDIR" \
   -Dbasedir="$BASEDIR" \
-  org.verapdf.apps.GreenfieldGuiWrapper \
-  "--frameScale" "$SCALE_FACTOR" \
-  "$@"
+  org.verapdf.apps.GreenfieldCliWrapper \
+  "${@}"
